### ISSUE TITLE
buildbot v2 REST API doesn't work with buildername

### ISSUE THREAD

**Issue by aakashja** [buildbot/buildbot#3407]
In buildbot 0.8 REST API, we could fetch a builder information using builder name, e.g.: 
https://buildbot.buildbot.net/json/builders/db-postgres?as_text=1

but in 0.9 REST API, we can’t fetch a builder information by builder-name:
http://nine.buildbot.net/api/v2/builders/buildbot-try (doesn't work)

Instead, we must know builder id before making the REST API call. builder id is assigned internally by buildbot. builder-id is not usually known beforehand to other applications (e.g. custom dashboard) fetching data from buildbot.

buildbot REST API should allow fetching information by builder name. It should have API of type:
 
/builders/i:buildername
/builders/i:buildername/builds

**Comment by tardyp** [comment]
@aakashja see #3410 for a fix. The full fix with buildernames in the UI implies more change to the message queues, which could impact performance. So they will need even more care.

**Comment by aakashja** [comment]
Thanks for the fix. I filed https://github.com/buildbot/buildbot/issues/3465 for the remaining UI issue. It would be great if it can be fixed as well, so that people do not have to remember the builder id and can easily type the URL based on builder names.

**Comment by tardyp** [comment]
People really want to type the full url with buildername and buildnumber?

**Comment by aakashja** [comment]
Yes, me and many people I know use this **all the time**. Once you open a particular builder/url, it's in browser history. So we can re-open the url from the history very quickly, knowing the builder name. With the builder id we can't do that.

Also, if the builder name follow certain naming convention, it's easy to switch between configurations/OS/version etc. by simply modifying url (e.g.: change 'debug' to 'release' in url).
